homeassistant/components/drop_connect/entity.py

```diff
@@ -0,0 +1,29 @@
+"""Base entity class for DROP entities."""
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import DROPDeviceDataUpdateCoordinator
+
+
+class DROPEntity(CoordinatorEntity[DROPDeviceDataUpdateCoordinator]):
+    """Representation of a DROP device entity."""
+
+    _attr_force_update = False
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self, entity_type: str, coordinator: DROPDeviceDataUpdateCoordinator
+    ) -> None:
+        """Init DROP entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.device_id}_{entity_type}"
+        self._attr_device_info = DeviceInfo(
```

Comment:
Are you recommending moving that definition into a property like this?

`@property`
`    def device_info(self) -> DeviceInfo:`
`        """Return a device description for device registry."""`
`        return DeviceInfo(`
`            identifiers={(DOMAIN, self.coordinator.config_entry.unique_id or "")},`
`            manufacturer=self.coordinator.manufacturer,`
`            model=self.coordinator.model,`
`            name=self.coordinator.device_name,`
`        )`

I believe I had it as a separate property at one point and moved it into the initializer on advice from another reviewer. I'm guessing that I'm misunderstanding because that wouldn't really remove any code.

Reply:
It does make sense that the DeviceInfo definition should be in the coordinator.  It's been moved there in 37285a815e3bdac717e2c19beb555edb576d4d33. (part of the refactor to add he device linkage)